api/src/main/java/org/pmiops/workbench/api/util/SQLGenerator.java

```diff
@@ -143,38 +187,25 @@ public QueryRequest handleSearch(String type, List<SearchParameter> params) {
         return returnParameters;
     }
 
-    /*
-     * The format placeholders stand for, in order:
-     *  - the table prefix (e.g., something like "pmi-drc-api-test.synpuf")
-     *  - the table name
-     *  - the table prefix again
-     *  - the *_SOURCE_CONCEPT_ID column identifier for that table
-     *  - the *_SOURCE_VALUE column identifier for that table
-     *  - a BigQuery "named parameter" indicating the list of codes to search by
-     */
-    private static final String subqueryTemplate = 
-        "SELECT DISTINCT PERSON_ID " +
-        "FROM `%s.%s` a, `%s.CONCEPT` b "+
-        "WHERE a.%s = b.CONCEPT_ID "+
-        "AND b.VOCABULARY_ID IN (@cm,@proc) " +
-        "AND a.%s IN (%s)";
-
-    protected String getSubQuery(String key) {
-        String tablePrefix = getTablePrefix();
+    private String getSubQuery(String key) {
         Map<String, String> info = tableInfo.get(key);
 
-        return String.format(subqueryTemplate, 
-            tablePrefix, 
-            info.get("tableName"),
-            tablePrefix, 
-            info.get("sourceConceptIdColumn"),
-            info.get("sourceValueColumn"),
-            "@" + key + "codes"
+        return String.format(SUBQUERY_TEMPLATE,
+                workbenchConfig.get().bigquery.projectId,
+                workbenchConfig.get().bigquery.dataSetId,
+                info.get("tableName"),
+                workbenchConfig.get().bigquery.projectId,
+                workbenchConfig.get().bigquery.dataSetId,
+                "CONCEPT",
+                info.get("sourceConceptIdColumn"),
+                "@" + key + "codes"
         );
     }
 
-    protected String getTablePrefix() {
-        // TODO: use the injectable config to grab this info
-        return "pmi-drc-api-test.synpuf";
+    private String setBigQueryConfig(String sqlStatement, String tableName) {
```

Comment:
Rather than having calls to this interspersed through the code, I think it would be better to have some token substitution thing that runs prior to execution... so e.g. ${projectId}.${dataSetId} gets converted to all-of-us-workbench-test.synpuf. There are a number of libraries we could use for this, or just use String.replace().

Reply:
(Don't have to change this now, but I think it would make things easier in future.)

Reply:
I may refactor how some of this works in the future. Waiting to see how it changes as we add in more criteria types.